packages/shared_preferences/shared_preferences/lib/shared_preferences.dart

```diff
@@ -18,8 +20,12 @@ class SharedPreferences {
   static const String _prefix = 'flutter.';
   static Completer<SharedPreferences> _completer;
 
-  static SharedPreferencesStorePlatform get _store =>
-      SharedPreferencesStorePlatform.instance;
+  static SharedPreferencesStorePlatform get _store {
+    if (LocalPlatform().isWindows) {
```

Comment:
I forgot that the registration of the default implementation is in the platform interface rather than this layer; this override has the unfortunate effect that setting a different instance explicitly (e.g., the in-memory implementation, or an unofficial alternate version) won't work on Windows.

I wonder if we should temporarily add a static property to disable this, so that in those cases people can get the dynamic behavior with, e.g., `SharedPreferencesStorePlatform.disablePlatformOverride = true;` before calling `getInstance()`. Removing it later when we don't need it would technically be a breaking change, but it would be a very easy one for people to deal with (just remove the line of code). @amirh, thoughts?